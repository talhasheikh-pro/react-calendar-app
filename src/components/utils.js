// common utils functions
import format from 'date-fns/format';

/**
 * 
 * @param {string} dateFrom 
 * @param {string} dateTo 
 * @returns array
 */
export function getAllDatesBetween(dateFrom, dateTo){
    const dates = [];
    for (let d = new Date(dateFrom); d < dateTo; d.setDate(d.getDate() + 1)) {
        dates.push(format(new Date(d), 'd LLL, Y'));
    }

    return dates;
}