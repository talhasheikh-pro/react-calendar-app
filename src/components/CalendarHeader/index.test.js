import renderer from 'react-test-renderer';
import Header from './index';
import add from 'date-fns/add';
import { getAllDatesBetween } from '../utils';

it('renders all dates', () => {
    const dayFrom = new Date();
    const to = add( dayFrom, {
        weeks: 1
    });

    const dates = getAllDatesBetween(dayFrom, new Date(to));
    const component = renderer.create(
        <Header dates={dates} />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});