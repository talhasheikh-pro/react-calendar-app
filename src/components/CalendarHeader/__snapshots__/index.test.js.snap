// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all dates 1`] = `
<section
  className="calendar__header"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-ee20y-MuiPaper-root"
  >
    27 Jul, 2022
     
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-ee20y-MuiPaper-root"
  >
    28 Jul, 2022
     
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-ee20y-MuiPaper-root"
  >
    29 Jul, 2022
     
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-ee20y-MuiPaper-root"
  >
    30 Jul, 2022
     
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-ee20y-MuiPaper-root"
  >
    31 Jul, 2022
     
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-ee20y-MuiPaper-root"
  >
    1 Aug, 2022
     
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-ee20y-MuiPaper-root"
  >
    2 Aug, 2022
     
  </div>
</section>
`;
